tensorbay/cli/utility.py

```diff
@@ -430,3 +437,28 @@ def wrapper(*args: Any, **kwargs: Any) -> None:
             error(str(err))
 
     return wrapper  # type: ignore[return-value]
+
+
+def sort_branches_or_tags(
+    sort_key: str, target: Union[PagingList[Branch], PagingList[Tag]]
+) -> Union[PagingList[Branch], PagingList[Tag], List[_NamedCommit]]:
```

Comment:
the signature is too complex, can we simplify it?
Also why we have `List[_NamedCommit]` in return type?

Reply:
The `sorted` function return `List[_NamedCommit]` type

Reply:
`PagingList` follows `Sequence` protocol, and `_NameCommit` is the base class of `Branch` and `Tag`,
so can the signature be like that? 
```python
def sort_branches_or_tags(
     sort_key: str, target: Sequence[_NameCommit]
 ) -> Sequence[_NamedCommit]:
```

or take one step further, can we use TypeVar to indicate that the input type is the same as output type:
```python
_T = TypeVar("_T", bound=_NameCommit) # or _T = TypeVar("_T", Tag, Branch)
def sort_branches_or_tags(
     sort_key: str, target: Sequence[_T]
 ) -> Sequence[_T]:
```